e2e/e2e.js

```diff
@@ -0,0 +1,177 @@
+
+    api.defaults.headers.Authorization = 'Bearer invalid-token';
+    await expect(api.get('/templates')).rejects.toThrow('Request failed with status code 401');
+    api.defaults.headers.Authorization = `Bearer ${config.authToken}`;
+
+    await expect(api.get('/templates/non-existent-id')).rejects.toThrow('Request failed with status code 404');
+  });
+
+  test('Verify authentication and authorization', async () => {
+    const unauthenticatedApi = axios.create({ baseURL: config.baseUrl });
+    await expect(unauthenticatedApi.get('/templates')).rejects.toThrow('Request failed with status code 401');
+
+    // Test with invalid token
+    const invalidApi = axios.create({
+      baseURL: config.baseUrl,
+      headers: { Authorization: 'Bearer invalid-token' }
+    });
+    await expect(invalidApi.get('/templates')).rejects.toThrow('Request failed with status code 401');
+
+    // Test with insufficient permissions
+    await expect(api.post('/some-restricted-endpoint')).rejects.toThrow('Request failed with status code 403');
+  });
+
+  test('Get template by organization and name', async () => {
+    const templateData = {
+      name: '@testorg/test-template',
+      links: { github: 'https://github.com/testorg/test-template' }
+    };
+    await createTemplate(templateData);
+
+    const getResponse = await api.get('/templates/testorg/test-template');
+    expect(getResponse.status).toBe(200);
+    expect(getResponse.data.name).toBe(templateData.name);
+
+    await expect(api.get('/templates/testorg/non-existent')).rejects.toThrow('Request failed with status code 404');
+  });
+
+  test('Verify concurrent template operations', async () => {
+    const createPromises = Array(5).fill().map(() => createTemplate({ name: `@adobe/concurrent-test-${Date.now()}` }));
+    const createdTemplates = await Promise.all(createPromises);
+
+    const updatePromises = createdTemplates.map(template =>
+      api.put(`/templates/${template.id}`, { updatedBy: 'ConcurrentTest', description: `Updated ${Date.now()}` })
+    );
+    await Promise.all(updatePromises);
+
+    const deletePromises = createdTemplates.map(template => api.delete(`/templates/${template.id}`));
+    await Promise.all(deletePromises);
+
+    // Verify all templates were deleted
+    for (const template of createdTemplates) {
+      await expect(api.get(`/templates/${template.id}`)).rejects.toThrow('Request failed with status code 404');
+    }
+  });
+
+  test('Verify API performance under load', async () => {
+    const startTime = Date.now();
+    const numRequests = 50;
+
+    const requests = Array(numRequests).fill().map(() => api.get('/templates'));
+    const responses = await Promise.all(requests);
+
+    const endTime = Date.now();
+    const totalTime = endTime - startTime;
+    const avgResponseTime = totalTime / numRequests;
+
+    console.log(`Average response time: ${avgResponseTime}ms`);
+
+    expect(avgResponseTime).toBeLessThan(3000); // Assuming 3000 ms is acceptable
```

Comment:
I would extract out this 3000 value as a constant at the top, perhaps as a config as well. So it's not hidden